astropy/time/core.py

```diff
@@ -1272,15 +1272,15 @@ def _get_delta_tdb_tt(self, jd1=None, jd2=None):
             rxy = np.hypot(location.x, location.y)
             z = location.z
             self._delta_tdb_tt = erfa.dtdb(
-                jd1, jd2, ut, lon.to(u.radian).value,
-                rxy.to(u.km).value, z.to(u.km).value)
+                jd1, jd2, ut, lon.to_value(u.radian),
+                rxy.to_value(u.km), z.to_value(u.km))
 
         return self._delta_tdb_tt
 
     def _set_delta_tdb_tt(self, val):
         val = self._match_shape(val)
         if hasattr(val, 'to'):  # Matches Quantity but also TimeDelta.
-            val = val.to(u.second).value
+            val = val.to_value(u.second)
```

Comment:
Good call on checking for other classes. This one is wrong (and the other setter too). Will correct.